jumanji/environments/routing/tsp/conftest.py

```diff
@@ -31,10 +36,59 @@ def sparse_reward() -> SparseReward:
 @pytest.fixture
 def tsp_dense_reward(dense_reward: DenseReward) -> TSP:
     """Instantiates a TSP environment with dense rewards and 5 cities."""
-    return TSP(num_cities=5, reward_fn=dense_reward)
+    return TSP(generator=RandomGenerator(num_cities=5), reward_fn=dense_reward)
 
 
 @pytest.fixture
 def tsp_sparse_reward(sparse_reward: SparseReward) -> TSP:
     """Instantiates a TSP environment with sparse rewards and 5 cities."""
-    return TSP(num_cities=5, reward_fn=sparse_reward)
+    return TSP(generator=RandomGenerator(num_cities=5), reward_fn=sparse_reward)
+
+
+class DummyGenerator(Generator):
+    """Hardcoded `Generator` mainly used for testing and debugging. It deterministically
+    outputs a hardcoded instance with 5 cities.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(
+            num_cities=5,
+        )
+
+    def __call__(self, key: PRNGKey) -> State:
+        """Call method responsible for generating a new state. It returns a travelling salesman
+        problem instance without any visited cities.
+
+        Args:
+            key: jax random key for any stochasticity used in the generation process. Not used
+                in this instance generator.
+
+        Returns:
+            A TSP State.
+        """
+        del key
+
+        coordinates = jnp.array(
+            [[0.0, 0.0], [0.0, 1.0], [1.0, 0.0], [1.0, 1.0], [0.5, 0.5]]
+        )
```

Comment:
Could we make the type explicit? To avoid any weak types. Though I understand this is for testing.
```suggestion
        coordinates = jnp.array(
            [[0.0, 0.0], [0.0, 1.0], [1.0, 0.0], [1.0, 1.0], [0.5, 0.5]],
            float,
        )
```